draft-ietf-taps-arch.md

```diff
@@ -108,6 +108,46 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+- Connection Group: A set of Connections that shares properties and caches.
+- Connection Property: A Transport Property that can be used to configure protocol-specific options and control per-connection behavior of a Transport Services implementation.
+- Endpoint: An identifier for one side of a transport connection.
+- Equivalent Protocol Stacks: Protocol stacks that can be safely swapped or raced in parallel during connection establishment.
+- Event: A primitive that is invoked by a transport endpoint {{?RFC8303}}.
+- Framer: A data translation layer that can be added to a Connection to define how application-layer Messages are transmitted over a transport stack.
+- Local Endpoint: A representation of the application's identifier for itself that it uses for transport connections.
+- Message:  A unit of data that can be transferred between two endpoints over a transport connection.
+- Message Property: A property than can be used to specify details about Message transmission.
+- Parameter: A value passed between an application and a transport protocol by a primitive {{?RFC8303}}.
+- Path: A representation of an available set of properties that a Local Endpoint can use to communicate with a Remote Endpoint.
+- Peer: An endpoint application party to a session.
+- Preconnection: an object that repeesents a Connection that has not yet been established.
+- Preference: A preference to prohibit, avoid, ignore prefer or require a specific feature.
+- Primitive: A function call that is used to locally communicate between an application and a transport endpoint.  A primitive is
+related to one or more transport features {{?RFC8303}}.
+- Protocol Instance: A single instance of one protocol, including any state necessary to establish connectivity or send and receive Messages.
+- Protocol Stack: A set of Protocol Instances that are used together to establish connectivity or send and receive Messages.
+- Racing: The attempt to select between multiple Protocol Stacks based on the Selection and Connection Properties communicated by the application, along with any security parameters.
+- Remote Endpoint: A representation of the application's identifier for a peer that can participate in a transport connection.
+- Rendezvous: The action of establishing a peer-to-peer connection with a Remote Endpoint.
+- Security Parameters: Parameters that define an application's requirements for authentication and encryption on a Connection.
+- Server: The peer responsible for responding to a session initiation.
+- Socket: The combination of a destination IP address and a destination port number {{?RFC8303}}..
+- System Policy: The input from an operating system or other global preferences that can constrain or influence how an implementation will gather candidate paths and Protocol Stacks and race the candidates during connection establishment.
```

Comment:
```suggestion
- System Policy: The input from an operating system or other global preferences that can constrain or influence how an implementation will gather candidate paths and Protocol Stacks and race the candidates during Connection establishment.
```

I keep fixing "connection" to "Connection" because in some cases I feel it is an important semantic difference. E.g., to me, "connection establishment" is the TCP handshake (in case of TCP), whereas "Connection establishment" is what a Transport System does after Initiate() is called, including racing etc.

Now, if we want to be super picky, we should probably also write "candidate Paths" here, since we're also writing "Protocol Stacks"...  maybe it would be better to write all of them in lower case though, as there won't be a misunderstanding about what they are anyway?